R/getUvalues.R

```diff
@@ -58,430 +49,129 @@ getUvalues <- function(config,
-    mutate(delta = .data[["value"]] - .data[["valueTarget"]]) %>%
-    select("region", "delta")
-
-  tmp <- df %>%
-    left_join(tmpDelta, by = "region") %>%
-    left_join(lambda, by = c("region", "period", "scenario")) %>%
-    mutate(value = ifelse(.data[["variable"]] == "uvalue",
-                          .data[["value"]] - .data[["delta"]] * (1 - .data[["fullconv"]]),
-                          .data[["value"]])) %>%
-    select(-"delta", -"lambda", -"fullconv")
-
-  return(tmp)
-}
-
-
-includeInputDataAndCorrect <- function(dfScen,
-                                       dfData,
-                                       gdppop,
-                                       lambda,
-                                       endOfHistory,
-                                       incomeThresholdUvalue,
-                                       multiUValue) {
-
-  dfData <- dfData %>% select("uvalue", "region", "period")
-
-  # Take the mean of the commercial and residential uvalues
-  dfData <- dfData %>%
-    group_by(across(all_of(c("region", "period")))) %>%
-    reframe(uvalue = mean(.data[["uvalue"]], na.rm = TRUE)) %>%
-    ungroup()
-
-  # Extend the historical data set
-  dfData <- dfData %>%
-    interpolate_missing_periods(period = getPeriods(dfScen),
-                                expand.values = TRUE,
-                                value = "uvalue")
-
-  tmpKeep <- dfScen %>%
-    filter(.data[["variable"]] != "uvalue") %>%
-    select("region", "period", "scenario", "variable", "value")
-
-  gdppop <- gdppop %>%
-    select("scenario", "period", "region", "variable", "value") %>%
-    spread(key = "variable", value = "value")
-
-  tmp <- dfScen %>%
-    filter(.data[["variable"]] == "uvalue") %>%
-    left_join(dfData, by = c("region", "period")) %>%
-    left_join(gdppop, by = c("scenario", "period", "region"))
-
-  tmp <- tmp %>%
-    left_join(lambda, by = c("region", "period", "scenario")) %>%
-    group_by(across(all_of("region"))) %>%
-    mutate(
-      uvalue = ifelse(is.na(.data$uvalue),
-                      .data$value * pmax(1, (multiUValue - 1) * (1 - .data$gdppop / incomeThresholdUvalue) + 1),
-                      .data$uvalue),
-      incSat = pmin(1, .data$gdppop / incomeThresholdUvalue),
-      factor = .data$incSat - .data$incSat[.data$period == endOfHistory],
-      uvalue = .data$uvalue * (1 - .data$factor / multiUValue),
-      delta = .data$uvalue - .data$value,
-      value = .data$value + .data$delta * (1 - .data$fullconv)) %>%
+    mutate(uvalueSmooth  = ifelse(.data$period >= endOfHistory, .data$uvalue,  lowpass(.data$uvalue, i = 5))) %>%
+    filter(!is.na(.data$uvalueSmooth)) %>%
+
+    # Find max historical value and period where it occurs
+    mutate(maxHistValue = max(.data$uvalueSmooth[.data$period <= endOfHistory], na.rm = TRUE),
+           periodMax    = .data$period[.data$period <= endOfHistory]
+           [which.max(.data$uvalueSmooth[.data$period <= endOfHistory])]) %>%
```

Comment:
Maybe easier to read
```suggestion
    mutate(hist         = .data$period <= endOfHistory,
           maxHistValue = max(.data$uvalueSmooth[hist], na.rm = TRUE),
           periodMax    = .data$period[hist][which.max(.data$uvalueSmooth[hist])]) %>%
```